### PR TITLE
add new implementation to run the nft list command

### PR DESCRIPTION
**Pull request by aabughosh**

### CODE REVIEW

File: test/e2e/nftables_test.go [hunk 1]
```diff
@@ -88,12 +93,12 @@ var _ = Describe("Nftables", func() {
 		}()
 
 		By("Listing nftables rules")
-		output, err := firewall.NftListAndWriteToFile(debugPod, utilsHelpers, artifactsDir, "nftables-after-reboot-"+nodeName)
```
Comment (sabinaaledort) [inline test/e2e/nftables_test.go:91]: If NftListAndWriteToFile is not used after this change, it can be removed

File: test/e2e/nftables_test.go [hunk 2]
```diff
@@ -88,12 +93,12 @@ var _ = Describe("Nftables", func() {
...
+		err = firewall.WriteToFile([]byte(logPod), utilsHelpers, artifactsDir, "nftables-after-reboot-"+nodeName)
```
Comment (sabinaaledort) [inline test/e2e/nftables_test.go:96]: No need to call firewall to write a file, can just do the write here

File: pkg/utils/utils.go [hunk 3]
```diff
@@ -142,8 +163,11 @@ func (u *utils) RunCommandOnPod(pod *corev1.Pod, command []string) ([]byte, erro
 	return buf.Bytes(), nil
 }
 
-func getPodDefinition(node string, namespace string, image string) *corev1.Pod {
+func getPodDefinition(node string, namespace string, image string, command []string) *corev1.Pod {
 	tolerationSeconds := int64(300)
+	if len(command) == 0 {
```
Comment (sabinaaledort) [inline pkg/utils/utils.go:168]: new line

File: pkg/utils/utils.go [hunk 4]
```diff
@@ -100,6 +101,28 @@ func (u *utils) CreatePodOnNode(nodeName, namespace, image string) (*corev1.Pod,
 	return pod, nil
 }
 
+func (u *utils) CreatePodOnNode(nodeName, namespace, image string) (*corev1.Pod, error) {
+	pod := getPodDefinition(nodeName, namespace, image, []string{})
+	return u.createPod(namespace, pod)
+}
+
+func (u *utils) CreatePodOnNodeWithCommand(nodeName, namespace, image string, command []string) (*corev1.Pod, string, error) {
```
Comment (sabinaaledort) [inline pkg/utils/utils.go:109]: The name of a func is CreatePodOnNodeWithCommand while it creates a pod and returns its log, it's not describing well what this func does, and anyway I don't think getting the log should be part of create pod func

File: pkg/utils/utils.go [hunk 5]
```diff
@@ -100,6 +101,28 @@ func (u *utils) CreatePodOnNode(nodeName, namespace, image string) (*corev1.Pod,
...
+	log.Print("on CreatePodOnNodeWithCommand")
+	pod := getPodDefinition(nodeName, namespace, image, command)
+	pod, err := u.createPod(namespace, pod)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to get pod: %w", err)
+	}
+	log.Print("pod on Created")
+
+	podLogs, err := u.getPodLogs(namespace, pod.Name)
```
[the remark this answers is not in the document]
Comment (aabughosh) [reply pkg/utils/utils.go:118]: its not on the wait for pod to be ready part?

File: pkg/utils/utils.go [hunk 6]
```diff
@@ -133,7 +151,7 @@ func (u *utils) RunCommandOnPod(pod *corev1.Pod, command []string) ([]byte, erro
 		Stdin:  os.Stdin,
 		Stdout: &buf,
 		Stderr: os.Stderr,
-		Tty:    true,
+		Tty:    false,
```
Comment (sabinaaledort) [inline pkg/utils/utils.go:154]: missing the change in VersionedParams we discussed

File: pkg/utils/utils.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (sabinaaledort) [reply pkg/utils/utils.go]: Ready state is before complete state

File: pkg/utils/utils.go (hunk 6 above)
Comment (aabughosh) [reply pkg/utils/utils.go:154]: i dont think all those needed 
like we are running the command on create now and this change is adding weird warning on the logs

File: test/pkg/node/node.go [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+		for _, condition := range node.Status.Conditions {
+			if condition.Type == v1.NodeReady && condition.Status != v1.ConditionTrue {
+				return true
+			}
+		}
+		return false
+	}, timeout, interval).Should(
+		gomega.BeTrue(),
+		fmt.Sprintf("Node %s is still ready after %s", nodeName, timeout.String()),
```
Comment (sabinaaledort) [inline test/pkg/node/node.go:71]: This should be printed with log

File: pkg/utils/utils.go [hunk 8]
```diff
@@ -142,9 +142,11 @@ func (u *utils) RunCommandOnPod(pod *corev1.Pod, command []string) ([]byte, erro
 	return buf.Bytes(), nil
 }
 
-func getPodDefinition(node string, namespace string, image string) *corev1.Pod {
+func getPodDefinition(node string, namespace string, image string, command []string) *corev1.Pod {
 	tolerationSeconds := int64(300)
-
+	if len(command) == 0 {
```
Comment (sabinaaledort) [inline pkg/utils/utils.go:147]: Can you add spaces between the lines?

File: pkg/utils/utils.go [hunk 9]
```diff
@@ -238,3 +242,25 @@ func (u *utils) IsBMInfra() (bool, error) {
 
 	return infra.Status.PlatformStatus.Type == configv1.BareMetalPlatformType, nil
 }
+
+// getPodLogs fetches logs from a pod once it is running.
```
Comment (sabinaaledort) [inline pkg/utils/utils.go:246]: The comment has to be fixed, this is not checking the the pod status

File: test/e2e/nftables_test.go [hunk 10]
```diff
@@ -79,21 +82,39 @@ var _ = Describe("Nftables", func() {
...
+		By("Rebooting first node: " + nodeName + "and waiting for disconnect \n")
+		err = node.SoftRebootNodeAndWaitForDisconnect(utilsHelpers, cs, nodeName, testNS)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Waiting for node to be ready")
+		node.WaitForNodeReady(nodeName, cs)
+
+		By("Listing nftables rules")
+		command := []string{
+			"chroot", "/host", "/bin/bash", "-c", "nft list ruleset;",
```
Comment (sabinaaledort) [inline test/e2e/nftables_test.go:94]: `;` is needed?

File: test/pkg/node/node.go [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+	}
+
+	defer func() {
+		err := utilsHelpers.DeletePod(debugPod)
+		if err != nil {
+			log.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+		}
+	}()
+
+	rebootcmd := []string{"chroot", "/host", "reboot"}
```
Comment (sabinaaledort) [inline test/pkg/node/node.go:42]: The reboot command can be applied when creating the pod

**Pull request merged**